clang/lib/APINotes/APINotesWriter.cpp

```diff
@@ -19,11 +19,11 @@
 
 namespace clang {
 namespace api_notes {
-class APINotesWriter::Implementation {
-  template <typename T>
-  using VersionedSmallVector =
-      llvm::SmallVector<std::pair<llvm::VersionTuple, T>, 1>;
+template <typename T>
+using VersionedSmallVector =
+    llvm::SmallVector<std::pair<llvm::VersionTuple, T>, 1>;
```

Comment:
Why make this public?

Reply:
This is used in both `APINotesWriter` and `APINotesWriter::Implementation`.

Reply:
That is an internal type, it should have visibility to it already.

Reply:
We can't access `APINotesWriter::Implementation::VersionedSmallVector` from `APINotesWriter::addObjCContext` unless we declare `friend class APINotesWriter`.
I just did that, let me know if you think this is better 🙂